### ISSUE TITLE
application declaration - supported format - not clear

### ISSUE THREAD

**Issue by amelon** [Unitech/PM2#1584]
reading doc [app declaration](http://pm2.keymetrics.io/docs/usage/application-declaration/) I see ``Declaration via JS, JSON or JSON5 file``.

Nice, thus I started making my config file in JS like that

my_pm2_conf.js
````
var isProd = (process.env.NODE_ENV == 'production');

module.exports = {
  "apps" : [
    {
      "name"   : "my-app",
      "script" : "index.js",
      "watch": isProd ? false : ["server", "shared"]
    }
};
````

with following command line

````
pm2 start ./my_pm2_conf.js --no-daemon
````

But it doesn't work, as pm2 cli command manage ``js`` file as a script and not a conf :(. 

What do you mean with ``Declaration via JS file`` ?

**Comment by Unitech** [comment]
Yes this is true. For now with the current PM2 version you can only use some basic Javascript functions in application declaration file. The file extension must be .json but you can still put comments and use Js functions into it

The usage of object process.env in application declaration file will be available in the next PM2 release, 

thanks for reporting sir!
---

Documentation fixed: http://pm2.keymetrics.io/docs/usage/application-declaration/

Calling runInThisContext to acces to process object https://github.com/Unitech/PM2/commit/4e7ab91f63934ef4b985f8375ba5f52ccaa4946e

**Comment by calebmer** [comment]
@Unitech What's the status on `process.env` in application declaration files?
